### PR TITLE
[DLSP21] Update 03-3.md

### PR DESCRIPTION
**Pull request by jaylizzhn**

### CODE REVIEW

**Comment by Atcold** [comment]
Ouch… This lesson is already transcribed (see last year).
All it was necessary was the first 5 to 10 minutes.

**Review by Atcold — CHANGES_REQUESTED** [review]

**Comment by jaylizzhn** [comment]
Hi, I have added the part for the first 5~10 minutes(introduction of Diagram.net). Have a look!
Feel free to remove any redundant content.

File: docs/en/week03/03-3.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+<center>
+<img src="{{site.baseurl}}/images/week03/03-3/figure10.png" style="background-color:#DCDCDC;" /><br>
+</center>
+
+In this example, $x$ and $y$  are observations.
+
+In the half above, we feed the $x$ to a given encoder to get a prediction $\bar y$. This is called forward propagation.
+
+In the half below, we want to get the prediction $\bar x$ given observation $x$ and $y$. We keep doing gradient descent to make the network output as close as to $y$. This is called amortizing inference.
```
Comment (Atcold) [inline docs/en/week03/03-3.md:39]: We don't have $x$, right? We want to infer $\bar{x}$ given $y$.
Also, please replace every `x` and `y` with `\vect{x}` and `\vect{y}`.

File: docs/en/week03/03-3.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+## Spiral classification
```
Comment (Atcold) [inline docs/en/week03/03-3.md:41]: For this, please, remove the duplicate content (text and figures) and add a link to [last year's writeup](https://atcold.github.io/pytorch-Deep-Learning/en/week02/02-3/).
Sorry for the double work.

Comment (Atcold) [reply docs/en/week03/03-3.md:41]: Feel free to add any clarification you think it's missing from last year's version.

**Review by Atcold — CHANGES_REQUESTED** [review]
Almost there. A few more edits and we're done here.

File: docs/en/week03/03-3.md (hunk 1 above)
Comment (jaylizzhn) [reply docs/en/week03/03-3.md]: I usually use \vec{x} in latex. I just want to confirm here we use \vect{x} instead of \vec{x}, is that correct?

Comment (Atcold) [reply docs/en/week03/03-3.md]: https://github.com/Atcold/NYU-DLSP21/blob/master/docs/_layouts/custom.html#L7

Comment (jaylizzhn) [reply docs/en/week03/03-3.md]: fixed 👍

**Review by Atcold — CHANGES_REQUESTED** [review]
Add everything until https://youtu.be/EyKiYVwrdjE?t=754
Remove all images that are not used.

**Comment by jaylizzhn** [comment]
Fixed, I have removed all unused images and introduced that backpropagation can be used for inference. Is there anything else you would like me to add?

**Comment by Atcold** [comment]
> Add everything until https://youtu.be/EyKiYVwrdjE?t=754

File: docs/en/week03/03-3.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+We use "Delay" to denote the encoder(e.g., neural network).
+
+
+<center>
+<img src="{{site.baseurl}}/images/week03/03-3/figure10.png" style="background-color:#DCDCDC;" /><br>
+</center>
+
+In this example, $\vect{x}$ and $\vect{y}$  are observations.
+
+In the half above, we feed the $\vect{x}$ to a given encoder to get a prediction $\vect {\bar{y}}$. This is called forward propagation.
```
Comment (Atcold) [inline docs/en/week03/03-3.md:37]: `\bar{\vect{}}`, not the other way around.

**Review by Atcold — CHANGES_REQUESTED** [review]

**Comment by Atcold** [comment]
Please, add all the new content.

**Comment by jaylizzhn** [comment]
Sorry, I don't quite understand what you are specifically referring to. 
I thought I had already added all the new content(5~10th minutes).

**Comment by Atcold** [comment]
Sorry if I was unclear.

> Add everything until https://youtu.be/EyKiYVwrdjE?t=754

**Comment by jaylizzhn** [comment]
From 5:00 to 12:00 you covered the diagram topic. Then 12:00 to 12:34 you talked about the Twitter topic. I still don't understand what more I should add.

**Comment by Atcold** [comment]
Everything from 0:00 to 12:34.

**Comment by jaylizzhn** [comment]
Yes, I get it wrong sorry. I will add the first 5 minutes soon.

**Comment by Atcold** [comment]
No worries. I understand I was not clear. It's my bad, actually.
Yeah, there should be 3 heading, I guess, about the toolings.

The practicum content is mostly the same as last year.
You can say (instead of converting the title into a link), the following content is mostly the same, so here you can find a link, or something along these lines.

**Comment by jaylizzhn** [comment]
Finished, have a look. By the way, I converted the title for Typora and Notion to their official site links. If you don't like it, I can remove the links.

File: docs/en/week03/03-3.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+In the half above, we feed the $\vect{x}$ to a given encoder to get a prediction $\bar {\vect{y}}$. This is called forward propagation.
+
+In the half below, we want to get the prediction $\bar{\vect{x}}$ given observation $\vect{y}$. We keep doing gradient descent to make the network output as close as to $\vect{y}$. This is called amortizing inference. 
+
+Usually, we use backpropagation to compute the gradient, then we apply gradient descent with those computed values to train the model. This example shows that backpropagation is NOT only used during training. Backpropagation can also be used for inference.
+
+
+
+## Spiral Classification
+The following content is mostly the same, so here you can find what you need(https://atcold.github.io/pytorch-Deep-Learning/en/week02/02-3/)
```
Comment (Atcold) [inline docs/en/week03/03-3.md:53]: You forgot the square brackets for the link.

File: docs/en/week03/03-3.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+---
+lang-ref: ch.03-3             
+title: Spiral classification 
+lecturer: Alfredo Canziani 
+authors: Wenhao Li          
+date: 6 May 2021        
+---                    
+
+## [Typora](https://typora.io/)
+Typora is a useful tool to write markdown and latex. It is convenient to write paper and homework, and generating pdf file with Typora.
```
Comment (Atcold) [inline docs/en/week03/03-3.md:10]:
```suggestion
Typora is a useful tool to write markdown with the addition of formulae in LaTeX. It is convenient to write paper and homework, and generating pdf file with Typora.
```

File: docs/en/week03/03-3.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+## [Notion](https://www.notion.so/)
+<center>
+<img src="{{site.baseurl}}/images/week03/03-3/figure1.png" style="background-color:#DCDCDC;" /><br>
```
Comment (Atcold) [inline docs/en/week03/03-3.md:14]: Linked databases is the feature I was trying to stress about. So, perhaps we should use the following instead?
image

I think linking to [this tutorial](https://www.notion.so/Intro-to-databases-fd8cd2d212f74c50954c11086d85997e) would be beneficial. But we need to explain how this should be used to consume online content for keeping up with Deep Learning. I hope you can see my point.

A similar database can be used to keep a reading list for articles one wants to read with its summaries. See work done by [@y0b1byte](https://twitter.com/y0b1byte), namely [this](https://twitter.com/y0b1byte/status/1394394401489707011).

File: docs/en/week03/03-3.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+</center>
+Here you can place all your favorite stuff. This includes but is not limited to recipes, music, books, notes. Everything in one place, simple and powerful.
+
+
+## Diagram.net
+
+Diagrams.net is a great tool to draw neural network diagrams. Next we will introduce a few rules to make our diagrams more consistent with the ones in lecture.
```
Comment (Atcold) [inline docs/en/week03/03-3.md:21]: Missing link?

**Review by Atcold — CHANGES_REQUESTED** [review]

File: docs/en/week03/03-3.md (hunk 7 above)
Comment (jaylizzhn) [reply docs/en/week03/03-3.md]: I have added more details. Have a look! 
Thanks.

**Comment by jaylizzhn** [comment]
I have added more details. Have a look!
Thanks.

**Comment by Atcold** [comment]
Thanks!

**Pull request merged**